parsl/executors/workqueue/executor.py

```diff
@@ -352,12 +352,17 @@ def WorkQueueCollectorThread(collector_queue=multiprocessing.Queue(),
         # Successful task
         else:
             result = item["result"]
-            future_update, _ = deserialize_object(result["result"])
+            future_update = result["result"]
             logger.debug("Updating Future for Parsl Task {}".format(parsl_tid))
             if result["failure"] is False:
                 future.set_result(future_update)
             else:
-                future.set_exception(RemoteExceptionWrapper(*future_update))
+                future_fail = pickle.loads(future_update)
+                exc = RemoteExceptionWrapper(*future_fail)
```

Comment:
This bit of code is maybe more convoluted than it needs to be. RemoteExceptionWrapper is intended to transport exceptions through serialisation (which has been a problem for us in the past). But, this code creates that wrapper and then immediately unwraps it by calling `reraise`.

If you just want to set that exception you could call `reraise(t, v, tb)` (where those are the three parameters you're probably getting from `future_fail`) on line 363 without the RemoteExceptionWrapper